constance/templates/admin/constance/includes/results_list.html

```diff
@@ -21,9 +21,23 @@
             {{ item.form_field }}
             <br>
             <a href="#" onClick="
-              document.getElementById('{{ item.form_field.auto_id }}').{% if item.is_checkbox %}checked =
-                  {% if item.raw_default %} true {% else %} false {% endif %}
-                {% else %}value = '{{ item.default|escapejs }}'{% endif %}; return false;">Reset to default</a>
+              {% if item.is_datetime %}
+                d = new Date(
+                  {{ item.raw_default|date:"Y" }},
+                  {{ item.raw_default|date:"n" }} - 1,
+                  {{ item.raw_default|date:"j" }},
+                  {{ item.raw_default|time:"H" }},
+                  {{ item.raw_default|time:"i" }},
+                  {{ item.raw_default|time:"s" }});
```

Comment:
This is ignoring a possible timezone attached to the datetime object. E.g. if the configuration value has been added with one in the settings, we need to make sure to respect that when instantiating the JavaScript Date object. That's especially important since we use `get_format` a bit below which can -- if `USE_L10N` is enabled -- render a timezone aware datetime in the current active timezone -- which depending on the project setup can and in my experience often is different from the server time.

Django solves this for the admin datetime widget by passing in the [UTC offset from the server side](https://github.com/django/django/blob/3c447b108ac70757001171f7a4791f493880bf5b/django/contrib/admin/templates/admin/base.html#L14-L15) and then uses it when it [constructs a new Date object](https://github.com/django/django/blob/3c447b108ac70757001171f7a4791f493880bf5b/django/contrib/admin/static/admin/js/admin/DateTimeShortcuts.js#L43-L53). I think we need to follow the same approach here to prevent accidentally setting the wrong value.